artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/HierarchicalObjectRepository.java

```diff
@@ -484,17 +482,10 @@ public int compare(final String o1, final String o2) {
          } else if (!o1.contains(singleWord) && o2.contains(singleWord)) {
             return -1;
          } else if (o1.contains(singleWord) && o2.contains(singleWord)) {
-            String[] leftSplits = o1.split(quotedDelimiter);
-            String[] rightSplits = o2.split(quotedDelimiter);
-            for (int i = 0; i < leftSplits.length; i++) {
-               String left = leftSplits[i];
-               if (left.equals(singleWord)) {
-                  if (rightSplits.length < i || !rightSplits[i].equals(singleWord)) {
-                     return -1;
-                  } else {
-                     return +1;
-                  }
-               }
+            int o1Count = StringUtils.countMatches(o1, singleWord);
+            int o2Count = StringUtils.countMatches(o2, singleWord);
+            if (o1Count != o2Count) {
+               return o1Count - o2Count;
```

Comment:
The MatchComparator should take into account the position the `singleWord` wild card too, ie `foo.*.too.*` is more specific than `*.too` in a hierarchical sense.

Reply:
although the tests seem to cover that...

@msingermann ?

Reply:
Didn't know that it should behave that way. Let me change the code as it doesn't take it into account. (neither the tests)

Reply:
Just pushed the change to watch the wildcard position and keep the right order.